\iffalse

% curl http://ulms2019.nsdevil.com:10002/edit/lms/self/data/original_kiba/figures.zip -o /anup_files/FILES/personal/Thesis/mainmatter/3-Methodology/images/figures.zip; % login error so file not downloaded
% mv ~/Downloads/figures.zip /anup_files/FILES/personal/Thesis/mainmatter/3-Methodology/images/figures.zip;
% unzip /anup_files/FILES/personal/Thesis/mainmatter/3-Methodology/images/figures.zip -d "/anup_files/FILES/personal/Thesis/mainmatter/3-Methodology/images/"; 
% rm /anup_files/FILES/personal/Thesis/mainmatter/3-Methodology/images/figures.zip


Statistical Analyses to DO:
1. KIBA Score Distribution
2. Amino Acid counts (20 standards) -- For each protein sequence get total counts, sort by length -- (corresponding to one-hot)
3. Drug Sequence Counts similar to amino acid (Logarithmic-corresponding to one-hots)
4. 
Things to do:
1. System Block remake (remove calculation of KIBA Scores)
\fi \usetikzlibrary{arrows,automata}
\chapter{Methodology}

\section{System Overview}

\subsection{System Block}

A protein-drug interaction problem goes through various stages of data processing for building a prediction system in a Deep Learning Network. The systematic diagram is shown by Figure \ref{fig:system}. The data is collected by exploring the available internet sources and required dataset is downloaded for processing. From the data, the missing values are removed in data preprocessing to aid proper training. From the raw drugs and proteins profile, SMILES and FASTA sequences are extracted respectively. Now the features are extracted based on the sequence provided. Then, the features are fed into Deep Learning Algorithm as shown in \ref{fig:system2} where the training is performed to create the right prediction system. The training follows by testing and validation of data under different settings of protein and drug combinations.

\begin{figure}[H]
  % \def\svgwidth{\columnwidth}
  % \includesvg{mainmatter/3-Methodology/images/block.svg}
  \centering
  \includegraphics[width=.8\linewidth]{mainmatter/3-Methodology/images/system_block.pdf}
  \caption{System Block Diagram for Protein-Drug Prediction}
  \label{fig:system}
\end{figure}


\subsection{Data Collection}

The dataset is collected from open-internet database. Basically, there are three types of data required in this work: protein, drug and interaction sets. The UniProt Library has been used for extracting proteins features, PubChem for drug features, and NCBI for interaction scores. Additionally, PSI-BLAST is used to generate PSSM matrices for the protein features downloaded.

UniProt contains database of 173,281 proteins of human (Homo sapiens) (until 2019). The protein document consists of the taxonomic classification, identifiers to other databases for cross-linking,  molecular properties, related specific bioactivity, functional property, canonical and isoforms of protein sequence. The protein fasta sequence in particular is of interest to this research. An API can be used to download the available information.  
\url{https://www.uniprot.org/help/programmatic_access}

>O00311

\seqsplit{MEASLGIQMDEPMAFSPQRDRFQAEGSLKKNEQNFKLAGVKKDIEKLYEAVPQLSNVFKIEDKIGEGTFSSVYLATAQLQVGPEEKIALKHLIPTSHPIRIAAELQCLTVAGGQDNVMGVKYCFRKNDHVVIAMPYLEHESFLDILNSLSFQEVREYMLNLFKALKRIHQFGIVHRDVKPSNFLYNRRLKKYALVDFGLAQGTHDTKIELLKFVQSEAQQERCSQNKSHIITGNKIPLSGPVPKELDQQSTTKASVKRPYTNAQIQIKQGKDGKEGSVGLSVQRSVFGERNFNIHSSISHESPAVKLMKQSKTVDVLSRKLATKKKAISTKVMNSAVMRKTASSCPASLTCDCYATDKVCSICLSRRQQVAPRAGTPGFRAPEVLTKCPNQTTAIDMWSAGVIFLSLLSGRYPFYKASDDLTALAQIMTIRGSRETIQAAKTFGKSILCSKEVPAQDLRKLCERLRGMDSSTPKLTSDIQGHASHQPAISEKTDHKASCLVQTPPGQYSGNSFKKGDSNSCEHCFDEYNTNLEGWNEVPDEAYDLLDKLLDLNPASRITAEEALLHPFFKDMSL}

PubCHEM and CHEMBL are drug databases used for feature extraction of drug molecules. PubCHEM is a database containing 96,881,514 drug compounds and associates to each using CID identifier. It allows programmatic access and downloads of database text files. The SMILES structure provided by the PubChem library is used to generate features corresponding to each drug molecule. The properties associated with the molecule is explored using CHEMBL database using a programmatic api request provided. 

\url{https://pubchemdocs.ncbi.nlm.nih.gov/programmatic-access}, 
\url{https://chembl.gitbook.io/chembl-interface-documentation/web-services}. 

\iffalse
UniCHEM is a very useful repository that has been integrated with various web-services. The queries can be done from GET request and JSON text can be retrieved in response to the request. The database can be used to retrieve structure information, mappings, InChIKey and source information related to drug molecule. The InChIKey is used in this work to construct features for small drug molecule (aka ligand).   
\url{https://www.ebi.ac.uk/unichem/rest}
\fi

CHEMBL379218

PubCHEM CID 11314340

\seqsplit{CC1=C2C=C(C=CC2=NN1)C3=CC(=CN=C3)OCC(CC4=CC=CC=C4)N}



For the drug-target interaction (i.e. drug-protein interaction), KIBA scores were used \cite{Tang2013} instead of binary classification. Thus, a regression model was used to predict the drug and protein interaction. The KIBA score regression has two major advantages over binary classification: interaction strength of similarly interacting ligands-target (drugs-protein) can be compared and the bias problem of unknown interactions is refrained \cite{Tang2013,ozturk2018deepdta}. Higher score means that there is more strength of interaction between the two. We use \arabic{no_drugs} ligands as drugs and \arabic{no_proteins} human proteins as target for the prediction problem.

\begin{table}[H]
  \centering
  \caption{KIBA Score Table}
  \begin{tabular}{|l|l|l|l|l|}
    \hline
   
   & O00238 & O00311 & O00329 & O00418  \\ \hline
  CHEMBL10 & 3.518514 & 3.100002 & 4.0 & 3.6  \\ \hline
  CHEMBL102000 & NaN & NaN & NaN & NaN  \\ 
  \hline
  
  \end{tabular}
  \end{table}


Various components were used to form the prediction system. Protein interaction depends on its structural, chemical, molecular(related to H-bond) and electrostatic properties. The structural representation form basis for creating features in other properties. The primary canonical structure of protein-drug set are fed into interaction block. The interaction parameter is filtered accordingly to the filter type. Similarly, the drug feature set are created to be trained with the machine learning algorithm. Finally, after training the training dataset, cross validation of the model was done.

\section{Building Components of Features Processing}

The prediction system is built upon the feature extraction of proteins and drugs. \ref{fig:system2} shows the building components of the Input Vectors for feeding the deep learning network. The KIBA Prediction id done by feeding canonical protein fasta and drug smiles. The other features are constructed on their basis. PSSM matrix is constructed using PSSM matrices of human genome protein library from UniProt~\cite{UniProtConsortium2018} and the protein's FASTA. Two features, \acrfull{et} and \acrfull{pssmdt} are extracted from PSSM. From FASTA, Labelled encodings and \acrfull{rpt} matrix are created. From the SMILES, only labelled encodings is extracted.


\begin{figure}[H]\centering
  \includegraphics[width=0.9\linewidth]{mainmatter/3-Methodology/images/Algorithm.pdf}
  \caption{Schematic Block Diagram for Protein-Drug Prediction}
  \label{fig:system2}
\end{figure} 




  \input{mainmatter/3-Methodology/sections/kiba.tex}
  \input{mainmatter/3-Methodology/sections/pssm.tex}
  \input{mainmatter/3-Methodology/sections/deepnet.tex}