\setcounter{page}{1}
\chapter{Introduction}
\section{Background}
Treatment of diseases have mostly been associated with applying foreign medicinal components into human body. The rudimentary means of curing diseases has been growing with applied ayurbedas since the past. The chemical perspective of curing diseases slowly evolved into modern chemistry as drug facilities developed around the globe. The extensive research and documentation changed the world where people have come to trust fully in chemist's drugs to mitigate the ailments in the body. With the growing chemical status of the community, the need to develop better drugs and quick solutions is in high demand. The major reasons being the identification of new diseases and understanding the mechanism to cure such diseases.

~\acrfull{cadd} mechanism are the developing paradigms in bio-informatics ever since the "Next Industrial Revolution" possibilities started growing with computational power and technical human resources in biomedicine~\cite{Leelananda2016,Brown2017}. \acrfull{hts} was evolving eventually to find precedence over finding novel therapeutics. Based on the computational resources required,~\acrshort{cadd} can be classified into two general categories: Structure-based \acrshort{cadd} and Ligand-based~\acrshort{cadd}. Structure based \acrshort{cadd} relies on knowledge of target protein structure to calculate interaction energies for all the compounds tested whereas Ligand-based \acrshort{cadd} exploits similarities of known active and inactive molecules.

This work relates mostly to the Structure based \acrshort{cadd} and partly to Ligand-based \acrshort{cadd}. The parameters of our work can be associated with \begin{itemize}
    \setlength\parindent{24pt}
    \item Target Structure, De novo design 
    \item Ligand Structure, Ligand based virtual screening
\end{itemize}
Target Structure and Ligand Structures are the major parameters of the research. The de-novo design has not been explored yet but we can test the drug designs from Structure Generator\footnote{Structure Generator: The molecules which are highly active, readily synthesizable and devoid of undesirable properties are used to construct new possible drugs and can be tested with multiple targets.}. The other aspects of target identification -- Molecular Dynamics, Pharmacophore modeling, Ligand Docking, \acrfull{qsar} etc. are beyond the scope of this work. So, the predictions from the model can't be relied fully to conclude the results. The pharmacophore models should take the results from this work and make decisive conclusions. 

\iffalse
(52498, 254)
180244
\fi
The Dataset contains scores of the interaction of proteins and drugs based on \acrshort{kiba} scores. We use 52498 drugs from CHEMBL and 254 proteins from UniProt to get an interaction of 180244, by removing the unrecognized interactions. The interactions are based on KIBA score, collected from KEGG(​Kyoto Encyclopedia of Genes and Genomes)​ dataset \cite{Kanehisa2000}.

\iffalse

For citations, use the function \textbackslash cite. \cite{gowar1989power} The references file is the sample.bib one. Google Scholar provides almost all the references in LaTeX form.

To insert a footnote, use the following command. \footnote{This is a footnote.} When necessary to use a nomenclature, define it on the same page for a better organization. Don't create NSN (Non-sense nomenclatures).

For figures, tables, equations and further information, open the file "tips.tex". If what you need is not found there, Google it.

% Add nomenclature at each new used nomenclature
\fi

\section{Statement of Problem}
The simple technique of encoding the sequence information of drugs and proteins to identify if a drug will interact with the protein or not has a major issue in that while drugs encoding information can be used to make drug related predictions, the protein encodings don't properly form their representational vectors. Therefore, modeling a machine learning algorithm sometimes overfit the situation or poorly classify the problem. In this work, we explore various techniques and reproduce a regression problem for solving the prediction problem.

\subsection{Selection of Prediction Score}

Out of the many score functions; STITCH, Davis, Metz\_Anastassiadis and KIBA scores, we found that the prediction of drug and protein interaction problem is convenient with KIBA scores. Again, KIBA scores database consists of experimental data and secondary data (from literature) of drug-target interaction. Choosing the KIBA as the output score for two protein and drug sequences, we model our machine learning algorithm by following a proper feature encoding technique.

\subsection{Selection of Features}

For the protein family, the focus here is with the kinase target family because of its essential roles in cellular signaling transduction for many cancers and inflammatory diseases. We concentrate on proteins dataset, specifically because their interaction is quite tricky when considered among chemical, atomic, structural and electrical nature of protein residues. Our basis for forming the matrices and vectors related to protein sequence comes from the fact that these feature sets represent specific properties related to the protein and its residues. Also, the literatures describing the feature sets characteristics and results motivates us towards the selection of these parameters: \acrshort{pssmdt}, \acrshort{edt}, \acrshort{rpt} and embedding vectors.


\section{Objectives}
The objectives of the research are:
\begin{itemize}
    \setlength\parindent{36pt}
    \item To determine the efficient different transformation matrices related to protein.
    \item To determine the right machine learning algorithm for modeling the protein-drug interactions.
\end{itemize}

% \section{Scope of Work}



\section{Organization of Report}

\subsection{Choosing Method of Interaction}
Out of the two methods of contact prediction: Global Methods and Local Methods, where Global Method tries to predict the label of one residue pair considering the label of others while Local Method tries to predict the label of one residue pair without considering the label of others; we use Global Methods as a means of contact prediction. We try to run different variations in Residual Methods: Using Distance Prediction, Folding, Coevolutionary features engineering. 

\subsection{Creating Analogy with Image}
For any protein sequence, instead of regarding them as segments, we try to run the whole protein sequence as an image: the residual contacts representing the pixels of the image.

\subsection{Deep Learning Network Selection}
Convnets, as they still are quite helpful in solving an image recognition problem, we used their variations to understand the performance with protein drug set. As a higher level of optimization problem, we use LSTM to create different components of Model Selection. 

\subsection{Training and Testing}
A basic PC was used to create initial models. A server with 4 CPUs was then used thence after the models were selected for training. The testing was done in normal PC for validation.
