@article{Jeong2011,
abstract = {While genome sequencing projects have generated tremendous amounts of protein sequence data for a vast number of genomes, substantial portions of most genomes are still unannotated. Despite the success of experimental methods for identifying protein functions, they are often lab intensive and time consuming. Thus, it is only practical to use in silico methods for the genome-wide functional annotations. In this paper, we propose new features extracted from protein sequence only and machine learning-based methods for computational function prediction. These features are derived from a position-specific scoring matrix, which has shown great potential in other bininformatics problems. We evaluate these features using four different classifiers and yeast protein data. Our experimental results show that features derived from the position-specific scoring matrix are appropriate for automatic function annotation.},
author = {Jeong, Jong Cheol and Lin, Xiaotong and Chen, Xue Wen},
doi = {10.1109/TCBB.2010.93},
file = {:Users/anupadkh/Downloads/jongcheoljeong2011.pdf:pdf},
issn = {15455963},
journal = {IEEE/ACM Transactions on Computational Biology and Bioinformatics},
keywords = {Clustering,and association rules,classification,data mining,feature extraction or construction,mining methods and algorithms},
mendeley-groups = {Protein{\_}Drug{\_}Interaction},
number = {2},
pages = {308--315},
title = {{On position-specific scoring matrix for protein function prediction}},
volume = {8},
year = {2011}
}
